test/unified_memory_allocation/uma_pools/disjoint_pool_config.cpp

```diff
@@ -0,0 +1,325 @@
+    ASSERT_EQ(configsMap.at(DESCRIPTORS[0]).limits->MaxSize, 16 * 1024 * 1024);
+
+    // test for device
+    ASSERT_EQ(configsMap.at(DESCRIPTORS[1]).MaxPoolableSize, 4 * 1024 * 1024);
+    ASSERT_EQ(configsMap.at(DESCRIPTORS[1]).Capacity, 4);
+    ASSERT_EQ(configsMap.at(DESCRIPTORS[1]).SlabMinSize, 64 * 1024);
+    ASSERT_EQ(configsMap.at(DESCRIPTORS[1]).limits->MaxSize, 16 * 1024 * 1024);
+
+    // test for shared
+    ASSERT_EQ(configsMap.at(DESCRIPTORS[2]).MaxPoolableSize, 0);
+    ASSERT_EQ(configsMap.at(DESCRIPTORS[2]).Capacity, 0);
+    ASSERT_EQ(configsMap.at(DESCRIPTORS[2]).SlabMinSize, 2 * 1024 * 1024);
+    ASSERT_EQ(configsMap.at(DESCRIPTORS[2]).limits->MaxSize, 16 * 1024 * 1024);
+
+    // test for read-only shared
+    ASSERT_EQ(configsMap.at(DESCRIPTORS[3]).MaxPoolableSize, 4 * 1024 * 1024);
+    ASSERT_EQ(configsMap.at(DESCRIPTORS[3]).Capacity, 4);
+    ASSERT_EQ(configsMap.at(DESCRIPTORS[3]).SlabMinSize, 2 * 1024 * 1024);
+    ASSERT_EQ(configsMap.at(DESCRIPTORS[3]).limits->MaxSize, 16 * 1024 * 1024);
+}
+
+TEST(disjointPoolConfigTests, disjointPoolConfigStringInvalidEnabledBuffers) {
+    // test for a string with invalid parameter for EnabledBuffers--
+    // (it should be set to a default)
+    std::string config = "-5;32M;host:0,3,2m,4m;device:1M,4,64k";
+    usm_pool::PoolConfigurations<typename DisjointPool::Config> configsMap =
+        usm_pool::getPoolConfigurationsFor<DisjointPool>(config);
+
+    // test for host
+    ASSERT_EQ(configsMap.at(DESCRIPTORS[0]).MaxPoolableSize, 0);
+    ASSERT_EQ(configsMap.at(DESCRIPTORS[0]).Capacity, 3);
+    ASSERT_EQ(configsMap.at(DESCRIPTORS[0]).SlabMinSize, 2 * 1024 * 1024);
+    ASSERT_EQ(configsMap.at(DESCRIPTORS[0]).limits->MaxSize, 32 * 1024 * 1024);
+
+    // test for device
+    ASSERT_EQ(configsMap.at(DESCRIPTORS[1]).MaxPoolableSize, 1 * 1024 * 1024);
+    ASSERT_EQ(configsMap.at(DESCRIPTORS[1]).Capacity, 4);
+    ASSERT_EQ(configsMap.at(DESCRIPTORS[1]).SlabMinSize, 64 * 1024);
+    ASSERT_EQ(configsMap.at(DESCRIPTORS[1]).limits->MaxSize, 32 * 1024 * 1024);
+
+    // test for shared
+    ASSERT_EQ(configsMap.at(DESCRIPTORS[2]).MaxPoolableSize, 0);
+    ASSERT_EQ(configsMap.at(DESCRIPTORS[2]).Capacity, 0);
+    ASSERT_EQ(configsMap.at(DESCRIPTORS[2]).SlabMinSize, 2 * 1024 * 1024);
+    ASSERT_EQ(configsMap.at(DESCRIPTORS[2]).limits->MaxSize, 32 * 1024 * 1024);
+
+    // test for read-only shared
+    ASSERT_EQ(configsMap.at(DESCRIPTORS[3]).MaxPoolableSize, 4 * 1024 * 1024);
+    ASSERT_EQ(configsMap.at(DESCRIPTORS[3]).Capacity, 4);
+    ASSERT_EQ(configsMap.at(DESCRIPTORS[3]).SlabMinSize, 2 * 1024 * 1024);
+    ASSERT_EQ(configsMap.at(DESCRIPTORS[3]).limits->MaxSize, 32 * 1024 * 1024);
+}
+
+TEST(disjointPoolConfigTests, disjointPoolConfigStringTooManyParameters) {
+    // test for when too many parameters are passed-- (the extra parameters
+    // should be ignored)
+    std::string config = "1;32M;host:0,3,2m,4;device:1M,4,64k,5";
```

Comment:
Add a test case where some parameter is doubled, like: "1;32M;host:0,3,2m,4;device:1M,4,64k,5;host:1,8,4m,100"?

Reply:
Done.